src/RequestProcessor.js

```diff
@@ -7,45 +7,30 @@ const { processRequest } = require('./request');
 const logger = require('../src/config/winston');
 const RequestDao = require('./model/RequestDao');
 const DataDao = require('./model/DataDao');
-const checkNodeConnection = require('./utils/checkNodeConnection');
 const oracleAbi = require('./config/abi/oracle.abi');
 
 
 class RequestProcessor {
-  constructor(oracleAddress) {
+  constructor(oracleAddress, blockEmitter) {
     this.oracleContract = new web3.eth.Contract(
       oracleAbi,
       oracleAddress,
     );
     this.requestDao = new RequestDao();
     this.dataDao = new DataDao();
+    this.blockEmitter = blockEmitter;
   }
 
-  async listen() {
-    const connected = await checkNodeConnection();
-    if (!connected) {
-      process.exit(1);
-    }
-
-    this.oracleContract.events.DataRequested()
-      .on('data', async (event) => {
-        logger.info(event);
-
-        const request = Object.assign({}, event.returnValues, { startedAt: Date.now() });
-
-        this.requestDao.saveRequest(request);
-      })
-      .on('error', err => logger.error(err));
-
-    this.oracleContract.events.DelayedDataRequested()
-      .on('data', async (event) => {
-        logger.info(event);
+  listen() {
+    this.blockEmitter.on('actualBlock', (blockNumber) => {
+      if (!this.lastProcessedBlock) {
```

Comment:
How about calling `this.processEvents` just once and defining a variable `const startBlock` using the ternary operator?